symengine/fields.cpp

```diff
@@ -462,4 +462,93 @@ GaloisFieldDict::gf_ddf_zassenhaus() const
     }
     return factors;
 }
+
+GaloisFieldDict
+GaloisFieldDict::_gf_pow_pnm1d2(const GaloisFieldDict &f,
+                                const integer_class &n,
+                                const std::vector<GaloisFieldDict> &b) const
+{
+    GaloisFieldDict f_in(f);
+    f_in %= *this;
+    GaloisFieldDict h, r;
+    h = r = f_in;
+    for (unsigned i = 1; i < n; ++i) {
+        h = h.gf_frobenius_map(*this, b);
+        r *= h;
+        r %= *this;
+    }
+    auto res = gf_pow_mod(r, (modulo_ - 1_z) / 2_z);
+    return res;
+}
+
+GaloisFieldDict GaloisFieldDict::gf_random(const integer_class &n,
```

Comment:
If `n` is not needed as a `integer_class`, it's better to accept a `long` or an `unsigned long`